src/System.Management.Automation/utils/StructuredTraceSource.cs

```diff
@@ -924,6 +924,61 @@ internal void WriteLine(string format, object arg1)
             }
```

Comment:
```
        internal void WriteLine(string format, object arg1)
        {
            if ((_flags & PSTraceSourceOptions.WriteLine) != PSTraceSourceOptions.None)
            {
                FormatOutputLine(
                    PSTraceSourceOptions.WriteLine,
                    writeLineFormatter,
                    format,
                    new object[] { arg1 });
            }
```

Since you are here, can you also add overloads to `FormatOutputLine` that takes 1 argument and 2 arguments respectively? We can save quite a bit of array allocation by doing that.

Reply:
FormatOutputLine calls StringBuilde.AppendFormat() and OutputLine() - both with "params object[] args". So I don't see how we could reduce allocations here in easy way.

Reply:
`StringBuilder.AppendFormat` has the following overloads:
```
AppendFormat(string format, object arg0)
AppendFormat(string format, object arg0, object arg1)
```
So we can have overloads of `FormatOutputLine` to accept 1 or 2 arguments.

Reply:
Maybe I was able to do what you ask.